plugins/redfish/tests/redfish.py

```diff
@@ -0,0 +1,125 @@
+        "@odata.id": "/redfish/v1/",
+        "RedfishVersion": "1.6.0",
+        "UUID": "92384634-2938-2342-8820-489239905423",
+        "UpdateService": {"@odata.id": "/redfish/v1/UpdateService"},
+    }
+    response = json.dumps(res, indent=4, separators=(",", ": "))
+    return Response(response=response, status=200, mimetype="application/json")
+
+
+@app.route("/redfish/v1/UpdateService")
+def update_service():
+
+    res = {
+        "@odata.id": "/redfish/v1/UpdateService",
+        "@odata.type": "#UpdateService.v1_8_0.UpdateService",
+        "FirmwareInventory": {
+            "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory"
+        },
+        "HttpPushUri": "/FWUpdate",
+        "HttpPushUriOptions": {
+            "HttpPushUriApplyTime": {
+                "ApplyTime": "Immediate",
+            }
+        },
+        "HttpPushUriOptionsBusy": False,
+        "ServiceEnabled": True,
+    }
+    response = json.dumps(res, indent=4, separators=(",", ": "))
+    return Response(response=response, status=200, mimetype="application/json")
+
+
+@app.route("/redfish/v1/UpdateService/FirmwareInventory")
+def firmware_inventory():
+
+    res = {
+        "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory",
+        "@odata.type": "#SoftwareInventoryCollection.SoftwareInventoryCollection",
+        "Members": [
+            {"@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/BMC"},
+            {"@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/BIOS"},
+        ],
+        "Members@odata.count": 2,
+    }
+    response = json.dumps(res, indent=4, separators=(",", ": "))
+    return Response(response=response, status=200, mimetype="application/json")
+
+
+@app.route("/redfish/v1/UpdateService/FirmwareInventory/BMC")
+def firmware_inventory_bmc():
+
+    res = {
+        "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/BMC",
+        "@odata.type": "#SoftwareInventory.v1_2_3.SoftwareInventory",
+        "Id": "BMC",
+        "LowestSupportedVersion": "1.30.367a12-rev1",
+        "Manufacturer": "Contoso",
+        "Name": "Contoso BMC Firmware",
+        "RelatedItem": [{"@odata.id": "/redfish/v1/Managers/BMC"}],
+        "ReleaseDate": "2017-08-22T12:00:00",
+        "SoftwareId": "1624A9DF-5E13-47FC-874A-DF3AFF143089",
+        "UefiDevicePaths": ["BMC(0x1,0x0ABCDEF)"],
+        "Updateable": True,
+        "Version": "1.45.455b66-rev4",
+    }
+    response = json.dumps(res, indent=4, separators=(",", ": "))
+    return Response(response=response, status=200, mimetype="application/json")
+
+
+@app.route("/redfish/v1/UpdateService/FirmwareInventory/BIOS")
+def firmware_inventory_bios():
+
+    res = {
+        "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/BIOS",
+        "@odata.type": "#SoftwareInventory.v1_2_3.SoftwareInventory",
+        "Id": "BIOS",
+        "LowestSupportedVersion": "P79 v1.10",
+        "Manufacturer": "Contoso",
+        "Name": "Contoso BIOS Firmware",
```

Comment:
I really hope this didn't come from a real system...

Reply:
Nah, I'm using Redfish-Mockup-Server as the reference data until real hardware arrives.